src/index.js

```diff
@@ -4,15 +4,17 @@ import "./index.css";
 import App from "./App";
 import { makeServer } from "./server";
 import { BrowserRouter } from "react-router-dom";
-
+import { ProductContextProvider } from "./context/ProductContext";
 // Call make Server
 makeServer();
 
 ReactDOM.render(
 	<React.StrictMode>
-		<BrowserRouter>
-			<App />
-		</BrowserRouter>
+		<ProductContextProvider>
+			<BrowserRouter>
```

Comment:
changes: we need to keep our context provider inside <BrowserRouter>, because <BrowserRouter> is provided by the package 'react-router-dom'. So our Providers should be inside packages given providers.

Reply:
implemented in new commit